openedx/core/djangoapps/content/search/api.py

```diff
@@ -497,6 +523,22 @@ def index_collection_batch(batch, num_done, library_key) -> int:
                 IncrementalIndexCompleted.objects.get_or_create(context_key=lib_key)
             status_cb(f"{num_collections_done}/{num_collections} collections indexed for library {lib_key}")
 
+            # Similarly, batch process Units in pages of 100
+            units = authoring_api.get_units(library.learning_package_id)
+            num_units = units.count()
+            num_units_done = 0
+            status_cb(f"{num_units_done}/{num_units}. Now indexing units in library {lib_key}")
```

Comment:
Should we loop directly through containers instead of units here to make it cleaner when we get to sections and subsections?

```suggestion
            containers = authoring_api.get_containers(library.learning_package_id)
            num_containers = units.count()
            num_containers_done = 0
            status_cb(f"{num_containers_done}/{containers_units}. Now indexing containers in library {lib_key}")
```